dialogue-core/src/test/java/com/palantir/dialogue/core/QueuedChannelTest.java

```diff
@@ -44,6 +44,11 @@
 @SuppressWarnings("FutureReturnValueIgnored")
 public class QueuedChannelTest {
 
+    private static final MetricName NUM_QUEUED_METRIC =
```

Comment:
would it be possible to have the generated code either make these `MetricName`s accessible or provide getters for the particular metrics? I think the latter is preferable

Reply:
We provide getters for everything except gauges, as always gauges are special in all sorts of dangerous ways.

Metric names would require similar builders to those we provide to register(gauge)/access(everything else) metrics, so exposing those directly doesn’t feel like the right complexity trade off.
Accessing a gauge is difficult. We can’t provide generic type information, and outside of testing we don’t want users to load a gauge<?> from the registry.
Thoughts?